strategy/from_json/date.ts

```diff
@@ -1,12 +1,13 @@
 // Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
 import { FromJSONStrategy } from "../compose_strategy.ts";
 import { JSONValue } from "../../serializable.ts";
+import { ERROR_INVALID_DATE } from "../../error_messages.ts";
 
 /** allows authors to pass a regex to parse as a date */
 export function createDateStrategy(regex: RegExp): FromJSONStrategy {
   return function _createDateStrategy(input: JSONValue): Date {
     if (typeof input !== "string" || !regex.exec(input)) {
-      throw new Error("invalid date");
+      throw new Error(ERROR_INVALID_DATE);
```

Comment:
I see we are only using the passed in regex for validation. Should you extract the 3 numbers and use that to build the date instead?

Reply:
Hey, 
There are probably some improvements that could be made here, but the goal is to parse a string and convert that to a Date if it passes. We're not sure what string or regex we'll get as well. Is there a bug it would be fixing?

Reply:
regex: `^Im going to shoot my foot$`
input: `Im going to shoot my foot`
it's gonna fail at `new Date(input)`
I agree it's not a main concern

Reply:
We can't fix issues in projects that use this library, and if they are inputting a "string like this" expecting a Date, we can reply on native JS to let them know they messed up.

Reply:
I have this test passing

```ts

test({
  name: "createDateStrategy - throws on invalid dates",
  fn() {
    class Test extends Serializable {
      @SerializeProperty({
        fromJSONStrategy: createDateStrategy(/^Im going to shoot my foot$/),
      })
      date!: Date;
    }
    try {
      new Test().fromJSON(`{"date":"Im going to shoot my foot"}`);
      fail("Non date string did not error");
    } catch (error) {
      assertEquals(error.message, ERROR_INVALID_DATE);
    }
  },
});
```